add notes and questions for bitcoin-inquisition-39

0xBEEFCAF3 opened the pull request:
First draft of questions and notes for [bitcoin-inquisition-39](https://github.com/bitcoin-inquisition/bitcoin/pull/39).

glozow commented on _posts/2024-03-06-bitcoin-inquisition-#39.md:
```diff
@@ -6,34 +6,57 @@ link: https://github.com/bitcoin-inquisition/bitcoin/pull/39
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+The following pull request (PR) reinstates the usage of OP_CAT in accordance with the specifications outlined in the draft BIP, which can be found [here](https://github.com/bitcoin/bips/pull/1525). Reviewers are encouraged to familiarize themselves with the BIP before examining the code.
```
Maybe offer some more background on the motivation for this PR? What can be built with it, etc.

glozow commented on _posts/2024-03-06-bitcoin-inquisition-#39.md:
```diff
@@ -6,34 +6,57 @@ link: https://github.com/bitcoin-inquisition/bitcoin/pull/39
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+1. OP_CAT is defined as 0x7e. Even after replacing an OP_SUCCESS opcode, libraries can continue to use 0x7e to represent concatenation operations. Why is this the case?
+
+1. A new script verify flag is introduced: `SCRIPT_VERIFY_OP_CAT`. Is this considered a consensus-validating flag or a policy?
+
+1. If it is policy, where does the mempool policy reject transactions that use OP_CAT? Why does this policy exist?
+
+1. In `deploymentinfo.cpp`, there are both an `OP_CAT` flag and a `DISCOURAGE_OP_CAT` flag. What is the rationale behind having both of these?
+
+1. When does consensus consider OP_SUCCESS126 replaced by OP_CAT?
+
+1. Why is it important to verify if OP_CAT is being executed in a non-segwitv0 or base-script context at L474 of `interpreter.cpp` rather than inside the opcode definition?
+
+1. This PR introduces new semantics for taproot-related script tests in `script_tests.json`. What issues or inefficiencies existed with the previous testing strategy?
+
+1. Are there any additional test cases you would like to see implemented that are not covered by the functional tests or the script tests in `script_tests.json`?
```
I'd recommend adding links to places in the code (from https://github.com/bitcoin-core-review-club/bitcoin/tree/prbitcoin-inquisition-39), as that can be a hint for how to answer the question and encourage people to do more code review. For example, you can link where `SCRIPT_VERIFY_OP_CAT` is defined/used.

glozow commented on _posts/2024-03-06-bitcoin-inquisition-#39.md:
```diff
@@ -6,34 +6,57 @@ link: https://github.com/bitcoin-inquisition/bitcoin/pull/39
 
+- The primary objective of this PR is to re-enable OP_CAT utilizing OP_SUCCESS semantics, replacing OP_SUCCESS126.
+  - OP_CAT is designed to concatenate two elements on the stack.
+  - If the resulting element is smaller than the `MAX_SCRIPT_ELEMENT_SIZE`, it is placed on the stack.
```
Similarly, it would be helpful to link to the lines of code you're describing

glozow reviewed (commented):
Thanks, these notes look really good! Just a few suggestions to add some color

glozow commented on _posts/2024-03-06-bitcoin-inquisition-#39.md:
```diff
@@ -6,34 +6,57 @@ link: https://github.com/bitcoin-inquisition/bitcoin/pull/39
 authors: [0xBEEFCAF3, EthanHeilman]
 components: ["consensus"]
-host: EthanHeilman
+host: EthanHeilman, arminsdev aka 0xBEEFCAF3
```
feel free to just change this to your own handle if you're hosting. the website has an index by host

0xBEEFCAF3 replied:
I'll be attending and co-hosting with Ethan. Happy to leave Ethan's name up.

glozow reviewed (commented):
@0xBEEFCAF3 lmk if you want to address comments. I can also add a commit to push up minor changes if you're ok with that. Thinking of merging this soon.

0xBEEFCAF3 commented:
> @0xBEEFCAF3 lmk if you want to address comments. I can also add a commit to push up minor changes if you're ok with that. Thinking of merging this soon.

Thanks for the review! Addressed the comments. Feel free to push any additional commits as you see fit

glozow commented on _posts/2024-03-06-bitcoin-inquisition-#39.md:
```diff
@@ -13,27 +13,59 @@ commit:
 
+1. OP_CAT is defined as 0x7e. Even after replacing an OP_SUCCESS opcode, libraries can continue to use 0x7e to represent concatenation operations. Why is this the case?
+
+1. A new script verify flag is introduced: [`SCRIPT_VERIFY_OP_CAT`](https://github.com/0xBEEFCAF3/bitcoin/blob/armin/re-enable-op-cat/src/script/interpreter.h#L162). Is this considered a consensus-validating flag or a policy?
```
for my own clarification: is this question asking whether this flag is on in {consensus, policy, both}?

0xBEEFCAF3 replied:
Its asking if its either or or both. Maybe it can be worded a bit better.

glozow reviewed (commented):
Ah sorry one last nit

glozow commented:
I deleted the template text and put your handle in quotes so it doesn't become 3203386099 (haha). Merging now. Thanks @0xBEEFCAF3 ! Looking forward to this :)

glozow commented on _posts/2024-03-06-bitcoin-inquisition-#39.md:
```diff
@@ -13,27 +13,59 @@ commit:
 
 _Notes and questions to follow soon!_
```
(I removed this in [f3abec0](https://github.com/bitcoin-core-review-club/website/pull/749/commits/f3abec0d1bcaec4315cfe5afb8647bc95261224d))
